pkg/sync/filepath_sync.go

```diff
@@ -90,3 +106,25 @@ func (fs *FilePathSync) Notify(ctx context.Context, w chan<- INotify) {
 	w <- &Notifier{Event: Event[DefaultEventType]{DefaultEventTypeReady}} // signal readiness to the caller
 	<-ctx.Done()
 }
+
+// yamlToJson is a generic helper function to convert
+// yaml to json
+func yamlToJson(rawFile []byte) (string, error) {
+	var ms map[string]interface{}
+	if err := yaml.Unmarshal(rawFile, &ms); err != nil {
+		return "", fmt.Errorf("unmarshal yaml: %w", err)
+	}
+
+	// Using jsoniter library here because json.Marshal doesn't understand keys of
+	// interface{} type (it can only understand string keys)
+	// More info:https://stackoverflow.com/q/35377477/6874596
+	var jsonit = jsoniter.ConfigCompatibleWithStandardLibrary
```

Comment:
I'm not sure about this dependency. We already unmarshal this structure in `pkg/eval/json_evaluator_model.go`, and the structure only has string keys.

Have you tried unmarshaling into a `map[string]interface{}`? I think that should be sufficient.

Reply:
The yaml parser interprets the nested structures as `map[interface{}]interface{}`, a solution for this is to either use a library or [convert the keys to string](https://stackoverflow.com/a/40737676/10039498)

Reply:
OK I understand the problem a bit better. I'd prefer to do a conversion ourselves instead of using a library though, but that's my opinion.

Interested in what others think is the best way to go.

Reply:
@skyerus thank you for explaining. @toddbaert basically what @skyerus said (SO link is in the code comment above). My comment might be difficult to understand. I will re-word it.

Reply:
> OK I understand the problem a bit better. I'd prefer to do a conversion ourselves instead of using a library though, but that's my opinion.

I guess doing the conversion ourselves would give us more control but I am not sure how valuable that would be since all we want to do is convert yaml to json (not to say we'd have to maintain that code + potential bugs in our code vs a popular library)

Reply:
> > OK I understand the problem a bit better. I'd prefer to do a conversion ourselves instead of using a library though, but that's my opinion.
> 
> I guess doing the conversion ourselves would give us more control but I am not sure how valuable that would be since all we want to do is convert yaml to json (not to say we'd have to maintain that code + potential bugs in our code vs a popular library)

Understood. I think I'd like to hear from @skyerus @AlexsJones @james-milligan and if they think the lib is fine I'm fine with that approach.

Reply:
I'm leaning towards doing the conversion ourselves (if it is as simple as the stackoverflow suggestion implies) to avoid the additional surface area exposed by introducing a new lib.

Reply:
Do you want me to make this change?

Reply:
Nevermind, I think the problem is we are using `gopkg.in/yaml.v2` in the above PR.
`gopkg.in/yaml.v3` seems to Unmarshal into map[string]interface{} for sub-maps too which should fix the issue without importing any library or writing the code for conversion.

Reply:
Example: https://play.golang.com/p/4iCzr8UmEHT

Reply:
I would prefer us to use `gopkg.in/yaml.v3`